### DISCUSSION TITLE
Diagnostic Tools Compatibility

### DISCUSSION THREAD

**Discussion by Falklan** [rnd-ash/ultimate-nag52-fw#69] [General]
Howdy again! I received a unit two days ago and I'm in the process of installing the device in my 2003 ML500. My battery voltage was a little low and it's being charged. I seem to be having a problem with my diagnostic tool recognizing the device is even connected.

I'm going to wait until the battery is fully charged and I shall check again.

Diagnostic tool: [2023 Launch Creader Elite BENZ](https://en.cnlaunch.com/products-detail/i-136.html)

**Comment by rnd-ash** [comment]
Hey, so I think this is one of the snags of this tcu. EGS52 (Stock TCU) will use either CAN or KLINE for diagnostics. This TCU will only support CAN (space limitations on the PCB - and Kline is only used on the stock EGS modules from 1998-2003, then 2003-2012 uses CAN). Your ML500 likely uses KLINE to talk to the OEM tcu for diagnostics, hence why your scanner won't see this one

**Comment by Falklan** [comment]
Okay now I understand why it's not visible to my tool. I'm checking into the OpenVehicleDiag project also. Now back to this truck.

Something is seriously wrong with this truck. It hit the 100k mile mark not long ago and I did a transmission service + conductor plate. Here it is months later, 3 replacement conductor plates (warranty) and a lot of time trying to troubleshoot this crazy thing. It surges when it shifts. When I'm on the throttle speeding up entering a highway or passing a vehicle, it just doesn't seem to know what to do since it downshift and then upshifts. After that, when I ease off the throttle, the damn thing downshifts and the kickdown can throw you through the windscreen/windshield and then shifts right back up to the next gear. 

Since I basically completed a transmission service, I need to warm the truck up and confirm the proper amount of ATF is present. I only had time to refill the ATF while the truck was cold. I put enough ATF to pass the cold check and tomorrow I will warm it to operating temperature and confirm proper amount at operating temperature. 

BTW, yes the low battery caused a few low voltage problems but I charged it for a while and it's back to normal.

I need to pull the transmission electrical harness and inspect it for faults. It seems like the signals from the speed sensors are possibly intermittent and the UNAG configuration app seems to have problems trying to recognize the presence of the speed sensors. I will post the error message I keep seeing later today since it's 4 am here.

**Comment by Falklan** [comment]
Yeah that Launch diagnostic tool supports CAN. ODBII compliant. Does your UNAG52 communicate using KLINE? I think not so what is this? How can I confirm which protocol the OEM TCM utilizes by the board configuration? 

The Launch CRE Benz also works with BMW, Audi, VW, Honda, Ford, Toyota etcetera etcetera etcetera. 

Okay I'm trying to wrap this around my small head. 3rd party diagnostic tools (topdon, launch, autel) are incapable of even reading from an open source module?

Also, I'm not positive about how these systems communicate. I'm a networking tech not a programmer or coder. I can read some code and understand it but it's quite limited. An example: When I use my diagnostic tool to run tests on the transmission, is the tool communicating with the TCM through the ECM? Or does the tool communicate directly through the ODBII connector with the TCM?

BTW the configuration app does recognize and read the speed sensors signals while I'm driving the truck. I don't know what I'm looking for in the live data to isolate this shifting problem it's experiencing.

**Reply by rnd-ash** [reply]
Ok, so to sum things up. The OEM tcu will use either KLINE or CAN to communicate, and so will your diagnostic scan tool. The issue is the cars ignition switch, which acts as a firewall. Early switches do not allow CAN data from the OBD port to be sent to the powertrain CAN network - where the tcu lives. So even if both the scanner and TCU support CAN diagnostics, the ignition switch firewall blocks it.

**Comment by Falklan** [comment]
BAM you nailed that explanation! Indeed good sir!

**Reply by Falklan** [reply]
IMMO gatekeeper?

I ask because a new ignition starter switch is $20.

**Reply by rnd-ash** [reply]
You won't be able to swap it for a new one. It's the physical design that's different. The earliest ignition switch I'm aware of that supports CAN diagnostics on the powertrain bus is EZS211 (specifically designed for the w211 platform). The design of the ML ignition switch is what's limiting here. Even if you get a "new" one, it would still have the same PCB design thus same limitations

**Reply by Falklan** [reply]
👍

**Reply by Falklan** [reply]
I know exactly what you mean here. The exact same reason I can't integrate a HUD. Yeah I understand and this is ONE of THOSE trucks...
